### DISCUSSION TITLE
Simple operator parser

### DISCUSSION THREAD

**Discussion by xsebek** [foonathan/lexy#116] [Q&A] [answered]
Hi, I would like to use `lexy::dsl::literal_set` to parse binary operators.

Used directly, it does not produce a value though:
```cpp
struct binary_operator {
    static constexpr auto rule =
             dsl::literal_set(
                    LEXY_LIT("=="),
                    LEXY_LIT("!="),
                    LEXY_LIT(">"),
                    LEXY_LIT(">="),
                    LEXY_LIT("<"),
                    LEXY_LIT("<="));
    static constexpr auto value = lexy::callback<BinaryOperator>(
        []() -> BinaryOperator {
            throw std::runtime_error("How am I supposed to use no arguments?");
        });
};
```
https://godbolt.org/z/K3h361xqq

Is there some magic trick to make a literal set produce a value? Ideally something like `dsl::better_lit<EQ, "==">`. :slightly_smiling_face: 

I was able to work around this using `dsl::identifier(dsl::ascii::punct)`, but that seems kind of hacky.

---

I found the calculator example, which uses `lexy::dsl::op` and inheritance from `lexy::dsl::infix_op_left`... and it is kind of hard to see how it even works. Also, it would be a total overkill to use it to parse my simple format:
```
condition := variable_name comparison_operator integer
```

The format is from [Advent of Code 2017 day 8](https://adventofcode.com/2017/day/8) and the whole example looks like this:
```
b inc 5 if a > 1
a inc 1 if b < 5
c dec -10 if a >= 1
c inc -20 if c == 10
```

The literal set would also be handy for the `inc`/`dec` literals.

**Comment by foonathan** [comment, marked as answer]
> Ideally something like dsl::better_lit<EQ, "==">

That's called `lexy::symbol_table<T>.map<LEXY_SYMBOL("==")>(EQ) and used together with `dsl::symbol`: https://lexy.foonathan.net/reference/dsl/symbol/#symbol
